phpBB/includes/functions_acp.php

```diff
@@ -294,34 +320,125 @@ function build_cfg_template($tpl_type, $key, &$new_ary, $config_key, $vars)
 			$min = isset($tpl_type[1]) ? (int) $tpl_type[1] : false;
 			$max = isset($tpl_type[2]) ? (int) $tpl_type[2] : false;
 
-			$tpl = '<input id="' . $key . '" type="' . $tpl_type[0] . '"' . (( $min !== false ) ? ' min="' . $min . '"' : '') . (( $max !== false ) ? ' max="' . $max . '"' : '') . ' name="' . $name . '" value="' . $new_ary[$config_key] . '" />';
+			$tpl = [
+				'tag'		=> 'input',
+				'id'		=> $key,
+				'type'		=> $tpl_type[0],
+				'name'		=> $name,
+				'min'		=> $min !== false ? $min : '',
+				'max'		=> $max !== false ? $max : '',
+				'value'		=> $new_ary[$config_key],
+			];
 		break;
 
 		case 'dimension':
 			$min = isset($tpl_type[1]) ? (int) $tpl_type[1] : false;
 			$max = isset($tpl_type[2]) ? (int) $tpl_type[2] : false;
 
-			$tpl = '<input id="' . $key . '" type="number"' . (( $min !== false ) ? ' min="' . $min . '"' : '') . (( $max !== false ) ? ' max="' . $max . '"' : '') . ' name="config[' . $config_key . '_width]" value="' . $new_ary[$config_key . '_width'] . '" /> x <input type="number"' . (( $min !== '' ) ? ' min="' . $min . '"' : '') . (( $max != '' ) ? ' max="' . $max . '"' : '') . ' name="config[' . $config_key . '_height]" value="' . $new_ary[$config_key . '_height'] . '" />';
+			$tpl = [
+				'tag'		=> 'dimension',
+				'width' => [
+					'id'		=> $key,
+					'type'		=> 'number',
+					'name'		=> 'config[' . $config_key . '_width]',
+					'min'		=> $min !== false ? $min : '',
+					'max'		=> $max !== false ? $max : '',
+					'value'		=> $new_ary[$config_key . '_width'],
+				],
+				'height' => [
+					'type'		=> 'number',
+					'name'		=> 'config[' . $config_key . '_height]',
+					'min'		=> $min !== false ? $min : '',
+					'max'		=> $max !== false ? $max : '',
+					'value'		=> $new_ary[$config_key . '_height'],
+				],
+			];
 		break;
 
 		case 'textarea':
-			$rows = (int) $tpl_type[1];
-			$cols = (int) $tpl_type[2];
-
-			$tpl = '<textarea id="' . $key . '" name="' . $name . '" rows="' . $rows . '" cols="' . $cols . '">' . $new_ary[$config_key] . '</textarea>';
+			$tpl = [
+				'tag'		=> 'textarea',
+				'id'		=> $key,
+				'name'		=> $name,
+				'rows'		=> (int) $tpl_type[1],
+				'cols'		=> (int) $tpl_type[2],
+				'content'	=> $new_ary[$config_key],
+			];
 		break;
 
 		case 'radio':
-			$key_yes	= ($new_ary[$config_key]) ? ' checked="checked"' : '';
-			$key_no		= (!$new_ary[$config_key]) ? ' checked="checked"' : '';
-
 			$tpl_type_cond = explode('_', $tpl_type[1]);
-			$type_no = ($tpl_type_cond[0] == 'disabled' || $tpl_type_cond[0] == 'enabled') ? false : true;
+			$type_no = !(($tpl_type_cond[0] == 'disabled' || $tpl_type_cond[0] == 'enabled'));
```

Comment:
This don't matter much, but parenthesis here is redundant

Reply:
Cleaned it up even a tiny bit better ;)